### DISCUSSION TITLE
Using webpack's ProvidePlugin with mdx

### DISCUSSION THREAD

**Discussion by orazdow** [mdx-js/mdx#2039] [Q&A] [answered]
I set up ProvidePlugin to provide the import for Raviger router's [Link](https://kyeotic.github.io/raviger/link/) component with the following line in webpack.config.js: 
`new webpack.ProvidePlugin({Link: ['raviger', 'Link']})`

What I found was that only the .js files in my project where affected so I still needed to add `import {Link} from 'raviger';` inmy mdx components. 

Is there an option in the mdx loader that I can set to get this functionality? It's possible I'm not understanding webpack'c configuration also, but so far I can not get this import to work in mdx using ProvidePlugin.

**Comment by wooorm** [comment, marked as answer]
You can pass `Link` to `MDXContent`, or provide it. See [Using MDX](https://mdxjs.com/docs/using-mdx/).
I don’t think you need `ProvidePlugin`, but I don’t know webpack well, you might be better off asking them.
